src/ProducerEventProcessor/ProducerEventProcessor.cpp

```diff
@@ -555,6 +617,12 @@ void ProducerEventProcessorImpl::ProcessWarningInstrumentingWithUprobesEventAndT
   client_capture_event_collector_->AddEvent(std::move(event));
 }
 
+void ProducerEventProcessorImpl::ProcessThreadStateSliceCallstackAndTransferOwnership(
```

Comment:
This is not transferring the ownership of the pointer and hence it's leaking the `ThreadStateSliceCallstack`.

Reply:
How would that look like? Do I need to have another ProducerCaptureEvent in the map, such that I can transfer ownership there? Or do I not do transfer ownership at all (similar to the fullcallstack)?

Reply:
I completely changed the way we deal with callstacks. Should be better now.